Tasks/AzureRmWebAppDeployment/operations/PublishProfileUtility.ts

```diff
@@ -0,0 +1,53 @@
+import tl = require('vsts-task-lib/task');
+import { TaskParameters } from './TaskParameters';
+import fs = require('fs');
+import * as Constant from './Constants';
+
+var packageUtility = require('webdeployment-common/packageUtility.js');
+var parseString = require('xml2js').parseString;
+
+export class PublishProfileUtility {
+
+    private static async GetPublishProfileJsonFromFile(publishProfilePath: string): Promise<any> {
+        return new Promise((response, reject) => {
+            var pubxmlFile = packageUtility.PackageUtility.getPackagePath(publishProfilePath);
+            var publishProfileXML = fs.readFileSync(pubxmlFile);
+            parseString(publishProfileXML, (error, result) => {
+                if(!!error) {
+                    reject(error);
+                }
+                var propertyGroup = result && result.Project && result.Project.PropertyGroup ? result.Project.PropertyGroup : null;
+                if(propertyGroup) {
+                    for (var index in propertyGroup) {
+                        if (propertyGroup[index] && propertyGroup[index].WebPublishMethod[0] === "MSDeploy") {
+                            response(propertyGroup[index]);
+                        }
+                    }
+                }
+                reject(tl.loc('ErrorNoSuchDeployingMethodExists'));
+            });
+        });
+    }
+
+    public static async GetTaskParametersFromPublishProfileFile(taskParams: TaskParameters): Promise<any> {
+        try {
+            var publishProfileJs = await PublishProfileUtility.GetPublishProfileJsonFromFile(taskParams.PublishProfilePath);
+            if(!publishProfileJs.hasOwnProperty("MSDeployServiceURL")) {
```

Comment:
Can we move this `MSDeployServiceURL `in constant too?

Reply:
done